### ISSUE TITLE
Data loading issues while train  

### ISSUE THREAD

**Issue by Practcdi** [sergioburdisso/pyss3#16]
Hey ,

### [Note] : I have pandas dataframe contain  2 columns as  ,
1) Text 
2) Label
____

```
from sklearn.model_selection import train_test_split
x_train, x_test, y_train, y_test = train_test_split(x_data,
                                                    y_data ,
                                                    test_size = 0.2, 
                                                    shuffle=False)
```

`train () and fit()` **methods are not working**

here is a reference code 

![image](https://user-images.githubusercontent.com/58843833/135833228-55efaefc-1ffd-48a4-86e6-977d7641ad50.png)

How to fix it?

Thanks

**Comment by angrymeir** [comment]
Hey @Practcdi,

## TLDR
this is due to the case, that `fit/train` requires a list of strings instead of a DataFrame. (See function documentation [here](https://pyss3.readthedocs.io/en/latest/api/index.html#pyss3.SS3.fit))

Fix: give `x_train.values.tolist(), y_train` to `clf.train()`

## Bit more insights on why it does not work:
Following the respective code lines ([here](https://github.com/sergioburdisso/pyss3/blob/master/pyss3/__init__.py#L2344-L2347)):
```python
x_train, y_train = list(x_train), list(y_train)

if len(x_train) != len(y_train):
    raise ValueError("`x_train` and `y_train` must have the same length")
```

If you give a dataframe to the variable `x_train` of `shape = (535544, 1)` casting this to a list will only return the column names.
Thus the check will compare the following: `0 == 535544`
 
The check fails because before the comparison `x_train` and `y_train` are converted to lists. ([here](https://github.com/sergioburdisso/pyss3/blob/master/pyss3/__init__.py#L2344))

**Comment by Practcdi** [comment]
> Hey @Practcdi,
> 
> ## TLDR
> this is due to the case, that `fit/train` requires a list of strings instead of a DataFrame. (See function documentation [here](https://pyss3.readthedocs.io/en/latest/api/index.html#pyss3.SS3.fit))
> 
> Fix: pass `x_train.values.tolist(), y_train` to `clf.train()`
> 
> ## Bit more insights on why it does not work:
> Following the respective code lines ([here](https://github.com/sergioburdisso/pyss3/blob/master/pyss3/__init__.py#L2344-L2347)):
> 
> ```python
> x_train, y_train = list(x_train), list(y_train)
> 
> if len(x_train) != len(y_train):
>     raise ValueError("`x_train` and `y_train` must have the same length")
> ```
> 
> If you pass a dataframe to the variable `x_train` of `shape = (535544, 1)` casting this to a list will only return the column names. Thus the check will compare the following:
> 
> ```python
> if 1 != 535544:
>     raise ValueError("`x_train` and `y_train` must have the same length")
> ```


Thanks lot 😊 

How to dump model and check validation accuracy (confusion matrix or classification report) ?

**Comment by sergioburdisso** [comment]
@Practcdi Thanks for sharing this issue with us!

@angrymeir Thanks for taking care of it :muscle:, btw, what do you think of adding an extra check at the beginning of `fit/train`  throwing  an `ValueError` exception saying something like  "the `x_train` argument is expected to be a list of strings" when the provided `x_train` isn't a list of string. :thinking:

**Comment by angrymeir** [comment]
@sergioburdisso Hm unsure about that one because...
1. Where to start and where to end? Is it only `fit/train` that needs this kind of validation or also other methods (potentially all methods with user input because of consistency)?
2.  I think it's difficult to detect if a `x_train` can be casted to a list of strings without information loss. E.g. while `pandas.DataFrame` can't be casted, `pandas.Series` can be casted without issues, so it should stay a valid option?
3. Its well documented, stating exactly what the function expects.
